testing/test_doctest.py

```diff
@@ -475,6 +475,85 @@ def foo():
                                     "--junit-xml=junit.xml")
         reprec.assertoutcome(failed=1)
 
+    def _run_doctest_report(self, testdir, format):
+        testdir.makepyfile("""
+            def foo():
+                '''
+                >>> foo()
+                   a  b
+                0  1  4
+                1  2  4
+                2  3  6
+                '''
+                print('   a  b\\n'
+                      '0  1  4\\n'
+                      '1  2  5\\n'
+                      '2  3  6')
+            """)
+        return testdir.runpytest("--doctest-modules", "--doctest-report", format)
+
+    def test_doctest_report_udiff(self, testdir, format='udiff'):
```

Comment:
Oh now I see what you intended in `test_doctest_report_case_insensitive`. In this case I suggest using `parametrize` here instead:

```python
@pytest.mark.parametrize('format', ['udiff', 'UDIFF', 'uDiFf'])
def test_doctest_report_udiff(self, testdir, format):
```

And removing `test_doctest_report_case_insensitive`.